### ISSUE TITLE
Town builds unnecessary junctions on highway

### ISSUE THREAD

**Issue by kiwitreekor** [OpenTTD/OpenTTD#8506]
## Version of OpenTTD
1.10.3

## Expected result
Towns should not build unnecessary junctions on highway (disallowed construction of houses along roadtype and disabled to be built by towns)

## Actual result
![image](https://user-images.githubusercontent.com/24273113/103709980-2d45d600-4ff7-11eb-9896-dc231aa0d2fa.png)
Towns build unnecessary junctions

## Steps to reproduce
1. Build highway connected with town
2. Wait until town grows

**Comment by Eddi-z** [comment]
"Highway" is not a default road in OpenTTD, so you might want to update your "steps to reproduce" by which NewGRFs need to be loaded.

**Comment by kiwitreekor** [comment]
I used Rattroad 1.0.2 to reproduce it, but it is reproducible with any newgrf which includes a roadtype with flag bit 2 set and bit 4 cleared.

**Comment by glx22** [comment]
Does it happen for all town layouts ?

**Comment by kiwitreekor** [comment]
![image](https://user-images.githubusercontent.com/24273113/103731487-9b56c100-5028-11eb-9eb1-623894188e93.png)
2x2 layout
![image](https://user-images.githubusercontent.com/24273113/103731807-5bdca480-5029-11eb-84d9-e8a951ad72b1.png)
3x3 layout
![image](https://user-images.githubusercontent.com/24273113/103732001-d7d6ec80-5029-11eb-9252-8daa875ee40b.png)
Default layout

Yes, it happens.

**Comment by kiwitreekor** [comment]
I found that town can build roads after making junctions, I'm not sure whether this is intended or not.
In my opinion, It doesn't seem natural that a city can make a junction on a highway.

**Comment by glx22** [comment]
@kiwitreekor can you check if #8535 solves the issue ? (A preview is available for this PR): https://preview.openttd.org/pr8535/

**Comment by kiwitreekor** [comment]
#8535 fixes the issue, thanks!

**Comment by JGRennison** [comment]
For reference I added a road/tram type flag in my branch to prevent towns from modifying tiles of a particular road/tram type. https://github.com/JGRennison/OpenTTD-patches/commit/946442ab6be8897b2efa6157866feded7a27770f
This is a bit more restrictive than the change proposed in #8535 in that it also covers some other town road operations such as removing free ends.
This sort of mechanism could be an alternative way of solving this.

I had highways (i.e. dual carriageway one way roads of a highway road type) in mind when I added this, but there are no GRFs using this as far as I aware.
